_rules/heading-descriptive-b49b2e.md

```diff
@@ -29,17 +29,19 @@ This rule applies to any element with the [semantic role][] of heading that is e
 
 ## Expectation
 
-Each target element describes the topic or purpose of its [section of the content][].
+Each target element describes the topic or purpose of any [section of content][] [associated][] with it.
```

Comment:
I don't think this idea of there being three types of sections is correct. Nor that all of them need to be described by the first semantic heading inside them. Granted, WCAG is open-ended as to what the heading is actually supposed to describe, but this seems like a stretch. I've never tested this way, I know Deque doesn't test this way, Trusted Tester v5 doesn't seem to test this way... I'm curious to know if anyone does test headings like this.

This is inconsistent with how I think web accessibility is supposed to work. The purpose of semantics is to describe the structure this is conveyed through how the content is presented. The semantic structure isn't intended to be its own independent information structure. The way I understand this, the heading isn't supposed to describe the "programatic section", it is supposed to describe whatever is visually presented as a section, and so as long as the programatic section matches the visual section, the heading will also describe the programatic section. When the visuals don't match the semantics, 1.3.1 comes into play, where it's decided if the difference is significant enough to be non-conformant.

If you want more opinions on this, we can put it on the agenda for the next CG call.